.github/workflows/checks.yml

```diff
@@ -45,17 +45,11 @@ jobs:
       - name: Install dependencies
         run: npm ci
 
-      - name: Workspace lint
-        run: npx nx workspace-lint
-
       - name: Format check
         run: npx nx format:check --base=$NX_BASE --head=$NX_HEAD
 
-      - name: Lint affected --quiet
-        run: npx nx affected --base=$NX_BASE --head=$NX_HEAD --target=lint --parallel=3 --quiet
-
-      - name: Test affected
-        run: npx nx affected --base=$NX_BASE --head=$NX_HEAD --target=test --parallel=3 --ci --code-coverage
+      - name: Affected lint & tests
+        run: npx nx affected --base=$NX_BASE --head=$NX_HEAD --targets=lint,test --parallel=5 --quiet --cache --ci
```

Comment:
How about including the format as well? this will also have the advantage of having the result of all these tasks at every run instead of just stopping at the first failure (typically formatting) and then discovering other failures little by little...